tests/e2e/e2e_setup_test.go

```diff
@@ -527,23 +422,20 @@ func (s *IntegrationTestSuite) upgrade() {
 				5*time.Minute,
 				time.Second,
 			)
-			s.T().Logf("reached upgrade height on %s container: %s", s.valResources[curChain.meta.Id][i].Container.Name[1:], s.valResources[curChain.meta.Id][i].Container.ID)
+			s.T().Logf("reached upgrade height on %s container: %s", containerName, containerId)
 		}
 	}
 
 	// remove all containers so we can upgrade them to the new version
 	for _, chainConfig := range s.chainConfigs {
-		curChain := chainConfig
-		for valIdx := range curChain.validators {
+		for valIdx, val := range chainConfig.validators {
 			if _, ok := chainConfig.skipRunValidatorIndexes[valIdx]; ok {
 				continue
 			}
-
-			var opts docker.RemoveContainerOptions
-			opts.ID = s.valResources[curChain.meta.Id][valIdx].Container.ID
-			opts.Force = true
-			s.dkrPool.Client.RemoveContainer(opts)
-			s.T().Logf("removed container: %s", s.valResources[curChain.meta.Id][valIdx].Container.Name[1:])
+			containerName := val.validator.Name
+			err := s.containerManager.RemoveValidatorResource(chainConfig.meta.Id, containerName)
```

Comment:
is the validator.Name <> containerName equality documented somewhere? Seems pretty non-obvious to me

Reply:
Validator's name is what we provide to `RunValidatorResource`'s `containerName` argument: https://github.com/osmosis-labs/osmosis/blob/c7cd058c5d8cfd4ca0e90dcfc124663e7e0d6fc9/tests/e2e/e2e_setup_test.go#L225

I also added a few doc lines in https://github.com/osmosis-labs/osmosis/pull/1824/commits/7dbe534d0f4fdb2a481214983cb521bfc2439b15 about this